### ISSUE TITLE
Bug: Missing Dept. Name on Invoice for Daniil K. w/ ROW

### ISSUE THREAD

**Issue by TracyLinder** [cityofaustin/atd-data-tech#5389]
### When KontsevoyD creates invoices, the department name doesn't populate. This isn't happening with other users. 

- CTM Task # (SCTASK0126071), assigned to Robert w/ Tango

**Comment by Nadin-Nader** [comment]
I ran a few tests on this issue on PROD, TEST and DEV. Here is what i found:
On PROD, the Department name did not populate for any of the Invoices i ran.
RSN's: 12617152, 12623304, 12624332, 12623339, 12625919

On TEST, the department name did show up for the following RSN invoices.
RSN's: 12396982, 12400627

On DEV, the department name also showed.

I will create a TASK and submit it to CTM to fix the issue

**Comment by Nadin-Nader** [comment]
I have created **INC0087005** and submitted it to AMANDA CTM. I also gave deb a heads up that i was putting in an incident for it and will now provide her the Incident Number to Track

**Comment by Nadin-Nader** [comment]
@TracyLinder spoke to Robert and he shared with me his results after researching the problem:

> [1:59 PM] Licari, Robert
>     It appears that it is working as developed. The code was never updated to accommodate a default record for no sub code on a folder type. We need to update the report query to pull the record in the validLookup 52200 for lookup1 = 1 for defaults (null sub code on a folder type). Can I close the incident and create a task for you all to update the invoice report? I have already rewritten it once and can shrink my change to just address this issue

> ​[2:03 PM] Licari, Robert
>     This is the problem query that gets no data:      (SELECT validlookup.lookupstring2
>         FROM   validlookup
>         WHERE  validlookup.lookupcode = 52200
>                AND validlookup. lookup1 = folder.subcode
>                AND Upper(validlookup.lookupstring) = folder.foldertype) Department_Name,

He created a TASK for it (**SCTASK0126071**) and put it on TANGO's Board

**Comment by TracyLinder** [comment]
@Nadin-Nader please update. Thank you.

**Comment by Nadin-Nader** [comment]
Sent an email on 3/23 to CTM asking them to change the priority on the TASK and to see if they had an update. I sent a follow up email  today since i had not heard back and the priority was still a 4:

> From: Nader, Nadin 
> Sent: Friday, March 26, 2021 11:32 AM
> To: CTM Amanda Tango <CTMAmandaTango@austintexas.gov>
> Cc: Davidson, Deanna <Deanna.Davidson@austintexas.gov>
> Subject: RE: Upgrade the Priority for SCTASK0126071
> 
> Hi Everyone,
> 
> Hope you are all doing well. I was just checking up on this TASK ask it is still set to a priority 4 and at the very least it’s a 3 (a higher priority 3) or low Priority 2.
> 
> Can you change the priority, please? And if there is an update, please let me know.  TY 
> 
> From: Nader, Nadin 
> Sent: Tuesday, March 23, 2021 10:20 AM
> To: CTM Amanda Tango <CTMAmandaTango@austintexas.gov>
> Cc: Davidson, Deanna <Deanna.Davidson@austintexas.gov>
> Subject: Upgrade the Priority for SCTASK0126071
> 
> Hi Team,
> 
> Hope you are all doing well. I was reaching out to inquire about a TASK, specifically SCTASK0126071. I noticed that it was a Priority 4. Unfortunately this is not correct. It is supposed to be at the very least a 3 (a higher priority 3) or low Priority 2. 
> 
> I was wondering if you can change that priority to a 3 and put high visibility on this since these are invoices going out to customers and they do not have the Department name on them.
> 
> Thank you and I appreciate everything yall do. If you have any questions please reach out.
> 
> Respectfully,
> 
> Nadin A. Nader (He/Him/His)
> IT Business Systems Analyst
> Data and Technology Services

**Comment by TracyLinder** [comment]
Hi @Nadin-Nader. Is there an update on this one?

**Comment by Nadin-Nader** [comment]
@TracyLinder Robert has loaded the fix to TEST and i was able to print out the invoice and it had the department name on it.
 I will send Daniil and Paloma and email asking them to test this on TEST.

**Comment by Nadin-Nader** [comment]
Sent email to Paloma, Kim and Daniil to test this on TEST:

> From: Nader, Nadin 
> Sent: Tuesday, April 6, 2021 10:16 AM
> To: Perez, Kimberly <Kimberly.Perez@austintexas.gov>; Kontsevoy, Daniil <Daniil.Kontsevoy@austintexas.gov>
> Cc: Amayo-Ryan, Paloma <Paloma.Amayo-Ryan@austintexas.gov>
> Subject: AMANDA Invoice Department Name Missing
> Importance: High
> 
> Hi Everyone,
> 
> Hope you are all doing well. I was reaching out in regards to the issue Daniil and a few others were experiencing where Department name not showing on the Invoices. It seems that Robert from CTM applied the fix and it is now working. 
> 
> Please log onto TEST and verify. Once we have confirmation that the issue is fixed on TEST, we will send the fix over to PROD.
> 
> Respectfully,
> 
> Nadin A. Nader (He/Him/His)
> IT Business Systems Analyst
> Data and Technology Services
>

**Comment by Nadin-Nader** [comment]
@TracyLinder Got a response from Daniil and it looks like its fixed on TEST. I let Robert know to go ahead and promote it. He asked that we run it by the AOB first on Thursday to get it on PROD.

**Comment by TracyLinder** [comment]
@Nadin-Nader can you please follow-up with Robert and add an update here? Thank you.

**Comment by Nadin-Nader** [comment]
**UPDATE**
Sent an email and Chat to Robert to see if he has heard back from the AOB on the UAT they did on this. The UAT was/is to make sure this wont affect any other departments.

## **Status:**
Waiting on the AOB's UAT feedback and Robert to Deploy to PROD

**Comment by Nadin-Nader** [comment]
@TracyLinder Robert Deployed this to PROD and Daniil has UATd it on PROD. It works and i will close the issue.. See email below from Daniil.

> From: Kontsevoy, Daniil <Daniil.Kontsevoy@austintexas.gov> 
> Sent: Wednesday, May 5, 2021 12:20 PM
> To: Nader, Nadin <Nadin.Nader@austintexas.gov>; Perez, Kimberly <Kimberly.Perez@austintexas.gov>
> Cc: Amayo-Ryan, Paloma <Paloma.Amayo-Ryan@austintexas.gov>
> Subject: RE: AMANDA Invoice Department Name Missing
> 
> Good afternoon,
> Thank you for fixing it. I’ve just printed an invoice in PROD and it looks good now!

![PROD UAT.png](https://images.zenhubusercontent.com/5e8dfbfabf171441c3a5684e/fd5c453f-35bc-48ed-bf6b-a58b6fe231f6)
